### PR TITLE
add isgraphical

### PR DESCRIPTION
**Pull request by CarloLucibello**
see also #360

### CODE REVIEW

File: doc/basicmeasures.md [hunk 1]
```diff
@@ -202,3 +202,12 @@ egonet(g, v::Int, d::Int; dir=:out)
 
 Returns the subgraph of `g` induced by the neighbors of `v` up to distance `d`. If `g` is a `DiGraph` the `dir` optional argument specifies the edge direction the edge direction with respect to `v` (i.e. `:in` or `:out`) to be considered. This is equivalent to `induced_subgraph(g, neighborhood(g, v, d, dir=dir)).`
 
+### isgraphical
+```julia
+isgraphical(degs::Vector{Integer}
```
Comment (sbromberger) [inline doc/basicmeasures.md]: I think you're forgetting a closed parens here.

File: doc/build.jl [hunk 2]
```diff
@@ -163,14 +163,14 @@ Centrality measures implemented in *LightGraphs.jl* include the following:
 *LightGraphs.jl* implements some common random graph generators:
 
 {{erdos_renyi, watts_strogatz, random_regular_graph, random_regular_digraph,
-    random_configuration_model, stochastic_block_model,
+    random_configuration_model, barabasi_albert, stochastic_block_model,
```
Comment (sbromberger) [inline doc/build.jl]: Are these unrelated corrections?

File: doc/build.jl [hunk 3]
```diff
@@ -163,14 +163,14 @@ Centrality measures implemented in *LightGraphs.jl* include the following:
...
     StochasticBlockModel, make_edgestream}}
 
 ## Static Graphs
 *LightGraphs.jl* also implements a collection of classic graph generators:
 
 {{CompleteGraph, CompleteBipartiteGraph, CompleteDiGraph, StarGraph, StarDiGraph,PathGraph, PathDiGraph,
- WheelGraph, WheelDiGraph, BinaryTree, DoubleBinaryTree, RoachGraph}}
+ WheelGraph, WheelDiGraph, BinaryTree, DoubleBinaryTree, RoachGraph, Grid}}
```
Comment (sbromberger) [inline doc/build.jl]: Is this related?

File: doc/generators.md [hunk 4]
```diff
@@ -42,10 +42,19 @@ For directed graphs, allocates an $n \times n$ sparse matrix of boolean as an ad
 random_configuration_model(n::Int, k::Array{Int}; seed=-1)
 ```
 
-Creates a random undirected graph according to the [configuraton model](http://tuvalu.santafe.edu/~aaronc/courses/5352/fall2013/csci5352_2013_L11.pdf). It contains `n` vertices, the vertex `ì` having degree `k[i]`.
+Creates a random undirected graph according to the [configuraton model](http://tuvalu.santafe.edu/~aaronc/courses/5352/fall2013/csci5352_2013_L11.pdf). It contains `n` vertices, the vertex `i` having degree `k[i]`.
 
 Defining `c = mean(k)`, it allocates an array of `nc` `Int`s, and takes approximately $nc^2$ time.
 
+Make sure that `k` is a graphical sequence (see `isgraphical`), otherwise the function won't ever return.
```
Comment (sbromberger) [inline doc/generators.md]: Why not make this a precondition of the function itself, with an optional argument that bypasses the check if you know for sure that it's graphical?

File: doc/index.md [hunk 5]
```diff
@@ -172,6 +172,10 @@ Please include version numbers of all relevant libraries and Julia itself.
 - PRs should contain one logical enhancement to the codebase.
 - Squash commits in a PR.
 - Open an issue to discuss a feature before you start coding (this maximizes the likelihood of patch acceptance).
+- Minimize dependencies on external packages, and avoid introducing new dependencies. In general,
+    - PRs introducing dependencies on core Julia packages are ok.
+    - PRs introducing dependencies on non-core "leaf" packages (no subdependencies except for core Julia packages) are less ok.
+    - PRs introducing dependencies on non-core non-leaf packages require strict scrutiny and will likely not be accepted without some compelling reason (urgent bugfix or much-needed functionality).
```
Comment (sbromberger) [inline doc/index.md]: What are you changing here?

File: src/connectivity.jl [hunk 6]
```diff
@@ -283,3 +283,21 @@ the edge direction the edge direction with respect to `v` (i.e. `:in` or `:out`)
 to be considered. This is equivalent to `induced_subgraph(g, neighborhood(g, v, d, dir=dir)).`
 """
 egonet(g::SimpleGraph, v::Int, d::Int; dir=:out) = induced_subgraph(g, neighborhood(g, v, d, dir=dir))
+
+"""
+    isgraphical(degs::Vector{Integer}
```
Comment (sbromberger) [inline src/connectivity.jl]: close parens missing

File: doc/generators.md (hunk 4 above)
Comment (CarloLucibello) [inline doc/generators.md]: I tought about it, but the test is computationally heavier ( O(n^2))  than the generator itself, so I thought it is better to let the users do the check separately

File: doc/index.md (hunk 5 above)
Comment (CarloLucibello) [inline doc/index.md]: it is just the effect of a doc rebuild

File: doc/build.jl (hunk 2 above)
Comment (CarloLucibello) [inline doc/build.jl]: yes, I'm sorry, couldn't resist. This is a recently added function with no correspondence in the docs

File: doc/build.jl (hunk 3 above)
Comment (CarloLucibello) [inline doc/build.jl]: same here as above

File: doc/generators.md (hunk 4 above)
Comment (sbromberger) [inline doc/generators.md]: It can be an optional precondition (defaulting to `false`, if you'd like, though I think `true` is safer) as well as a separate check (so keep your code.

File: doc/index.md (hunk 5 above)
Comment (sbromberger) [inline doc/index.md]: Sounds like you didn't merge upstream first. You might want to do that.

File: doc/build.jl (hunk 2 above)
Comment (sbromberger) [inline doc/build.jl]: No worries. :)

**Comment by sbromberger** [comment]
@CarloLucibello mind getting the test coverage up?

**Comment by CarloLucibello** [comment]
For latest PR i cannot visualize the diff hit from codecov. Clicking on "Details" I get

No report found to at 078a6f9 (head) to compare against.

Any clue?

**Comment by sbromberger** [comment]
Any objections to me merging this and then getting you to check codecov and fix whatever's found? :)

**Comment by CarloLucibello** [comment]
go on!
